File: cakk-api/src/main/java/com/cakk/api/service/views/ViewsService.java
package com.cakk.api.service.views;

import static java.util.Objects.*;

import org.springframework.context.ApplicationEventPublisher;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import lombok.RequiredArgsConstructor;

import com.cakk.common.exception.CakkException;
import com.cakk.domain.mysql.entity.cake.Cake;
import com.cakk.domain.mysql.event.views.CakeIncreaseViewsEvent;
import com.cakk.domain.mysql.repository.reader.CakeReader;

@RequiredArgsConstructor
@Service
public class ViewsService {

	private final CakeReader cakeReader;

	private final ApplicationEventPublisher publisher;

	@Transactional(readOnly = true)
	public void increaseCakeViews(final Long cakeId) {
		if (isNull(cakeId)) {
			return;
		}

		try {
			final Cake cake = cakeReader.findById(cakeId);
			final CakeIncreaseViewsEvent event = cake.getInCreaseViewsEvent();

			publisher.publishEvent(event);
		} catch (CakkException ignored) {
			// ignored
		}
	}
}

File: cakk-core/src/main/kotlin/com/cakk/core/service/views/ViewsService.kt
package com.cakk.core.service.views

import org.springframework.context.ApplicationEventPublisher
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

import com.cakk.common.exception.CakkException
import com.cakk.core.facade.cake.CakeReadFacade
import com.cakk.core.mapper.supplyCakeIncreaseViewsEventBy

@Service
class ViewsService(
	private val cakeReadFacade: CakeReadFacade,
	private val publisher: ApplicationEventPublisher
) {

	@Transactional(readOnly = true)
	fun increaseCakeViews(cakeId: Long?) {
		cakeId?.let {
			try {
				val cake = cakeReadFacade.findById(cakeId)
				val event = supplyCakeIncreaseViewsEventBy(cake.id)

				publisher.publishEvent(event)
			} catch (ignored: CakkException) {
				// ignored
			}
		}
	}
}
